src/components/Brush.tsx

```diff
@@ -4,15 +4,16 @@ import { select } from "d3-selection"
 
 const flatten = (list) =>
   list.reduce(
-    (a, b) => a.concat(Array.isArray(b) ? flatten(b.sort((a, b) => a - b)) : b),
+    (a, b) =>
+      a.concat(Array.isArray(b) ? flatten([...b].sort((a, b) => a - b)) : b),
```

Comment:
I suggest to do `arr.slice()` instead of spreading because TypeScript may generate something that is far from being optimized

```suggestion
      a.concat(Array.isArray(b) ? flatten(b.slice().sort((a, b) => a - b)) : b),
```